src/parser/parser.py

```diff
@@ -0,0 +1,60 @@
+import slate
+
+
+import logging
+
+# import numpy
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Article:
+    """Class for storing articles.
+    """
+
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles 
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+    while i < len(pdf_text) - 1:
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
+            pdf_text[i + 1] == str(article_count) + ".-"
+            or pdf_text[i + 1] == str(article_count) + "-"
+            or pdf_text[i + 1] == str(article_count) + "."
```

Comment:
You did the testing but please do tell me, is "articulo 7" not good enough. I know there might be multiple instances of that string but you do have article_count. Do you still need to check .-, -, and .?

Reply:
usually it has the punctuation in the same token. will optimize later